cassandra/components.py

```diff
@@ -1161,3 +1160,390 @@ def run_component(self):
+
+    def build_data_list(self, rds_dict, target_file_dict, year_list):
+        """Generate a list of values for a given variable name and target scenario.
+
+        :param rds_dict:                A Python dictionary of {variable name: value arrays} derived from the emulator
+                                        RDS file.
+        :type rds_dict:                 dict
+
+        :param target_file_dict:        A dictonary of file names and their corresponding index from the
+                                        `infiles` variable from the RDS file that match the user defined
+                                        configuration parameters.
+                                        Format: {'files': ['<file path>'], 'file_index': [<file index>]}
+        :type target_file_dict:         dict
+
+        :param year_list:               A list of integer years that encompass the data range per realization.
+        :type year_list:                list
+
+        :return:                        A list of values for a target parameterization
+
+        """
+
+        # number of years for a realization
+        n_years = len(year_list)
+
+        # index of file corresponding to the position of the window containing the data for the target params
+        file_index = target_file_dict['file_index'][0]
+
+        # start and end index for the data window to extract
+        start_index = n_years * file_index
+        end_index = n_years * (file_index + 1)
+
+        return rds_dict[self.RDS_TGAV_NAME][start_index:end_index]
+
+    def build_dataframe(self, year_list, tgav_list):
+        """Build output data frame for Tgav.
+
+        :param year_list:               A list of integer years that encompass the data range per realization.
+        :type year_list:                list
+
+        :param tgav_list:               A list of values for a given variable name and target scenario
+        :type tgav_list:                list
+
+        :param target_file_dict:        A dictonary of file names and their corresponding index from the
+                                        `infiles` variable from the RDS file that match the user defined
+                                        configuration parameters.
+                                        Format: {'files': ['<file path>'], 'file_index': [<file index>]}
+        :type target_file_dict:         dict
+
+        :return:                        A data frame holding Tgav outputs and required ancillary data
+
+        """
+        import pandas as pd
+
+        # build a pandas data frame to hold tgav output
+        df = pd.DataFrame({'year': year_list, 'value': tgav_list})
+
+        # additional expected fields
+        df['scenario'] = self.params[self.SCENARIO_FIELD]
+        df['variable'] = self.TGAV_CAPABILITY_NAME
+        df['units'] = self.params[self.UNITS_FIELD]
+
+        return df[self.TGAV_FIELD_ORDER]
+
+    def tgav_metadata(self, df, target_file_dict):
```

Comment:
See my comments above about a better way to handle metadata.  I really think that insofar as possible, we need to generate this information down in the framework's classes.  We could potentially create a hook for a component to add extra metadata beyond what the framework can glean from the input configuration, but it should be optional.